pkg/cloudprovider/azure.go

```diff
@@ -134,27 +136,36 @@ func (a *Azure) initCredentials() error {
 		return fmt.Errorf("failed to initialize Azure environment: %w", err)
 	}
 
-	authorizer, err := a.getAuthorizer(a.env, cfg)
+	cred, cloudConfig, err := a.getAzureCredentials(a.env, cfg)
 	if err != nil {
 		return err
 	}
 
-	a.vmClient = compute.NewVirtualMachinesClientWithBaseURI(a.env.ResourceManagerEndpoint, cfg.subscriptionID)
-	a.vmClient.Authorizer = authorizer
-	_ = a.vmClient.AddToUserAgent(UserAgent)
+	options := &arm.ClientOptions{
```

Comment:
I am confused by this, all creds returned from `getAzureCredentials` already have the clientOptions set to this value. What's the point of doing it again here? Or is the one in getAzureCredentials redundant?

Reply:
Part of this changed in a newer commit from when this comment was made. Right now, I'm reusing the same cloudConfig generated in `getAzureCredentials`.  All of the clients have an options parameter so I'm making sure we are using the same cloudConfig in those as we used in `getAzureCredentials`.

Reply:
It is still unclear why are we using the same client options in creds and as a parameter here but I think it is fine for now.